openmc/mgxs_library.py

```diff
@@ -553,23 +585,526 @@ def nu_fission(self, nu_fission):
                 msg = "Invalid Shape of Nu_fission!"
                 raise ValueError(msg)
         else:
-            # Get shape of nu_fission so we can figure if we need chi or not
+            # Get shape of nu_fission to determine if we need chi or not
             if nu_fission.shape == shape_vec:
                 self._use_chi = True
-                shape = shape_vec
             elif nu_fission.shape == shape_mat:
                 self._use_chi = False
-                shape = shape_mat
             else:
                 msg = "Invalid Shape of Nu_fission!"
                 raise ValueError(msg)
 
         # check we have a numpy list
-        check_type("nu_fission", nu_fission, np.ndarray, expected_iter_type=Real)
+        check_type("nu_fission", nu_fission, np.ndarray,
+                   expected_iter_type=Real)
         self._nu_fission = np.copy(nu_fission)
         if np.sum(self._nu_fission) > 0.0:
             self._fissionable = True
 
+    def set_total(self, total, **kwargs):
+        if (isinstance(total, openmc.mgxs.TotalXS) or
+            isinstance(total, openmc.mgxs.TransportXS)):
+            # Make sure passed MGXS object contains correct group structure
+            if self.energy_groups != total.energy_groups:
```

Comment:
I *think* lines 608-611 can be reduced to one line as follows:

```python
cv.check_value('energy groups', total.energy_groups, self.energy_groups)
```

If this is indeed the case, it would of course apply to all of the other setters.